input.lisp

```diff
@@ -184,8 +184,8 @@ Available completion styles include
 ;;; keysym functions
 
 (defun is-modifier (keycode)
-  "Return t if keycode is a modifier"
-  (or (find keycode *all-modifiers* :test 'eql)
+  "Return T if KEYCODE is a modifier."
+  (or (member keycode *all-modifiers* :test 'eql)
       ;; Treat No Symbol keys as modifiers (and therefore ignorable)
```

Comment:
This comment implies a little smell: either the function should be named `ignorable-keycode-p`, or it should be named `modifier-p` and the calling site be checking for 0 separately.

Reply:
Yeah, I think all the modifier related code could be written in a more straightforward manner. This PR is part of a larger effort I'm trying to simplify it. For example why keep the mapping of keycode to modifer in a struct if I'm 'storing' the struct in a global variable. Using an alist makes more sense to me. Why do we not keep track of the control key? And so forth. But I still need to understand the current code in input.lisp better.